From: Jiri Slaby <jslaby@suse.cz>
Subject: Revert "net: fix a potential rcu_read_lock() imbalance in
 rt6_fill_node()"
Patch-mainline: never

This reverts commit e033155d0b495becfdc28f46ad20089dc8a13060, upstream
commit 94f826b8076e2cb92242061e92f21b5baa3eccc2.

We revert the patch which this fixes. So we need tto revert even this
one.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 net/ipv6/route.c |    8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/net/ipv6/route.c b/net/ipv6/route.c
index 8e600f8..e70e902 100644
--- a/net/ipv6/route.c
+++ b/net/ipv6/route.c
@@ -2411,12 +2411,8 @@ static int rt6_fill_node(struct net *net,
 
 	rcu_read_lock();
 	n = dst_get_neighbour(&rt->dst);
-	if (n) {
-		if (nla_put(skb, RTA_GATEWAY, 16, &n->primary_key) < 0) {
-			rcu_read_unlock();
-			goto nla_put_failure;
-		}
-	}
+	if (n)
+		NLA_PUT(skb, RTA_GATEWAY, 16, &n->primary_key);
 	rcu_read_unlock();
 
 	if (rt->dst.dev)
-- 
1.7.9.2

